pkg/webhook/clusterresourceplacement/v1beta1_clusterresourceplacement_mutating_webhook_test.go

```diff
@@ -0,0 +1,797 @@
+						Path:      "/spec/strategy/applyStrategy",
+						Value: map[string]any{
+							"comparisonOption": string(placementv1beta1.ComparisonOptionTypePartialComparison),
+							"type":             string(placementv1beta1.ApplyStrategyTypeClientSideApply),
+							"whenToApply":      string(placementv1beta1.WhenToApplyTypeAlways),
+							"whenToTakeOver":   string(placementv1beta1.WhenToTakeOverTypeAlways),
+						},
+					},
+					{
+						Operation: "add",
+						Path:      "/spec/revisionHistoryLimit",
+						Value:     float64(defaulter.DefaultRevisionHistoryLimitValue),
+					},
+				},
+			},
+		},
+		"should not patch if all fields present (UPDATE)": {
+			req: admission.Request{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					Name: "test-crp-all-fields",
+					OldObject: runtime.RawExtension{
+						Raw:    crpUpdateAllFieldsOldBytes,
+						Object: crpUpdateAllFieldsOld,
+					},
+					Object: runtime.RawExtension{
+						Raw:    crpUpdateAllFieldsNewBytes,
+						Object: crpUpdateAllFieldsNew,
+					},
+					UserInfo: authenticationv1.UserInfo{
+						Username: "test-user",
+						Groups:   []string{"system:masters"},
+					},
+					RequestKind: &utils.ClusterResourcePlacementMetaGVK,
+					Operation:   admissionv1.Update,
+				},
+			},
+			wantResponse: admission.Response{
+				AdmissionResponse: admissionv1.AdmissionResponse{
+					Allowed: true,
+				},
+				Patches: []jsonpatch.JsonPatchOperation{},
+			},
+		},
+		"should patch default if a field is changed (UPDATE)": {
+			req: admission.Request{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					Name: "test-crp-all-fields",
+					OldObject: runtime.RawExtension{
+						Raw:    crpUpdateChangeFieldOldBytes,
+						Object: crpUpdateChangeFieldOld,
+					},
+					Object: runtime.RawExtension{
+						Raw:    crpUpdateChangeFieldNewBytes,
+						Object: crpUpdateChangeFieldNew,
+					},
+					UserInfo: authenticationv1.UserInfo{
+						Username: "test-user",
+						Groups:   []string{"system:masters"},
+					},
+					RequestKind: &utils.ClusterResourcePlacementMetaGVK,
+					Operation:   admissionv1.Update,
+				},
+			},
+			wantResponse: admission.Response{
+				AdmissionResponse: admissionv1.AdmissionResponse{
+					Allowed:   true,
+					PatchType: ptr.To(admissionv1.PatchTypeJSONPatch),
+				},
+				Patches: []jsonpatch.JsonPatchOperation{
+					{
+						Operation: "add",
+						Path:      "/spec/policy/tolerations/0/operator",
+						Value:     string(corev1.TolerationOpEqual),
+					},
+					{
+						Operation: "add",
+						Path:      "/spec/strategy/applyStrategy/serverSideApplyConfig",
+						Value:     map[string]any{"force": bool(false)},
+					},
+				},
+			},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			resp := mutator.Handle(context.Background(), tc.req)
+			sortPatches(resp.Patches)
+			sortPatches(tc.wantResponse.Patches)
+			if diff := cmp.Diff(tc.wantResponse, resp); diff != "" {
```

Comment:
I think we usually use https://pkg.go.dev/github.com/google/go-cmp/cmp/cmpopts#SortSlices